packages/frontend/pages/contribute.tsx

```diff
@@ -1,29 +1,57 @@
 import { useState } from "react";
 import type { NextPage } from "next";
 import { Toast, toast } from "react-hot-toast";
+import { hashMessage } from "viem";
+import { useAccount, useSignTypedData } from "wagmi";
 
 type ResultUrl = {
   url: string;
   success: boolean;
   error?: string;
 };
 
+const EIP_712_DOMAIN = {
+  name: "Collective DAO Catalog",
+  version: "1",
+  chainId: 10,
+} as const;
+
+const EIP_712_TYPES = {
+  Message: [{ name: "urlHash", type: "string" }],
+} as const;
+
 const Contribute: NextPage = () => {
   const [url, setUrl] = useState<string>("");
   const [urls, setUrls] = useState<string>("");
   const [isSubmitting, setIsSubmitting] = useState(false);
 
+  const { signTypedDataAsync } = useSignTypedData();
+  const { address } = useAccount();
+
   const onSubmitUrl = async () => {
     try {
       setIsSubmitting(true);
 
-      // TODO: add signature
+      let signature;
+      try {
+        signature = await signTypedDataAsync({
+          domain: EIP_712_DOMAIN,
+          types: EIP_712_TYPES,
+          primaryType: "Message",
+          message: { urlHash: hashMessage(JSON.stringify([url])) },
```

Comment:
@carletex Why the URLs are hashed?

Reply:
@damianmarti Just because it's a common practice (I guess because it doesn't grow in size even if you submit 100 URLs, in case you want to store the signature somewhere).

But we could do it without hashing...

Reply:
I'm not sure. I like to be able to see what I'm signing and maybe it's most clear if the users can see the URLs they are sending, but it's not a big deal.

Reply:
Yes, I agree. Let's merge and tweak if necessary (it'd be super easy, just removing the hash function on both sides)

Thanks Damu!!